lib/parser.ex

```diff
@@ -0,0 +1,198 @@
+defmodule Membrane.Ogg.Parser do
+  import Bitwise
+
+  defmodule Packet do
+    defstruct payload: <<>>,
+              track_id: 0,
+              bos?: false,
+              eos?: false
+  end
+
+  @type continued_packets_t :: %{integer => binary}
+
+  @spec parse(binary, %{}) ::
+          {parsed :: list(Packet), unparsed :: binary, continued_packets :: continued_packets_t}
```

Comment:
I would either use `%{}; []` or `map(), list()` in typespecs.  I would also suggest renaming `unparsed` to `rest` and putting it at the end of return tuple to be consistent with other parts of the code

```suggestion
  @spec parse(binary, continued_packets_t) ::
          {parsed :: [Packet.t()], continued_packets :: continued_packets_t, rest :: binary}
```